lib/src/components/NxModal/NxModal.tsx

```diff
@@ -4,147 +4,41 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React, {FunctionComponent, KeyboardEvent, useEffect, useRef, useState} from 'react';
+import React, {FunctionComponent} from 'react';
 import classnames from 'classnames';
 import { pick, omit } from 'ramda';
 
 import withClass from '../../util/withClass';
-import {Props, propTypes} from './types';
+import AbstractDialog, { AbstractDialogContext } from '../AbstractDialog/AbstractDialog';
 
-import './NxModal.scss';
+import { CloseHandler, Props, propTypes } from './types';
 
-export const NxModalContext = React.createContext<HTMLDialogElement | null>(null);
+import './NxModal.scss';
 
-// Typescript has rather obnoxiously partially removed support for HTMLDialogElement.
-// See https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029#issuecomment-907490128 and
-// https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029#issuecomment-968299542 . This even though
-// Firefox appears to intend to finally add support in v98. Until TS restores this, we'll have to jump
-// through some hoops...
-const hasWindow = typeof window !== 'undefined',
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    dynamicallyTypedWindow = hasWindow && window as any,
-    hasNativeModalSupport = !!(hasWindow && dynamicallyTypedWindow.HTMLDialogElement &&
-      dynamicallyTypedWindow.HTMLDialogElement.prototype.showModal);
+export const NxModalContext = AbstractDialogContext;
```

Comment:
This probably shouldn't be done.  Everywhere that uses the context should use the new name.  It's not exported publicly so it's not a backwards compatibility concern.

Reply:
> This probably shouldn't be done. Everywhere that uses the context should use the new name. It's not exported publicly so it's not a backwards compatibility concern.

Fixed as part of this commit: e9dc0853371ec473190542de724a2867c8bc7264